pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -77,8 +80,82 @@ func (pl *CSILimits) Name() string {
 func (pl *CSILimits) EventsToRegister() []framework.ClusterEventWithHint {
 	return []framework.ClusterEventWithHint{
 		{Event: framework.ClusterEvent{Resource: framework.CSINode, ActionType: framework.Add}},
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}},
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}, QueueingHintFn: pl.isSchedulableAfterPodDeleted},
+	}
+}
+
+func (pl *CSILimits) isSchedulableAfterPodDeleted(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	deletedPod, _, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		logger.V(5).Info("Unexpected objects in isSchedulableAfterPodDeleted", "oldObj", oldObj, "err", err)
+		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterPodDeleted: %w", err)
+	}
+
+	if deletedPod.Namespace != pod.Namespace {
+		return framework.Queue, nil
 	}
+
+	if len(deletedPod.Spec.Volumes) == 0 {
+		return framework.Queue, nil
+	}
```

Comment:
Should be QueueSkip?

Reply:
1efeff94d8cc1694153d26569ace6f0f2221d1e4
3174d008109adfd01acfec9c0bdf291894372d76